lib/adf_diag.py

```diff
@@ -915,4 +933,25 @@ def setup_run_cvdp(self):
         print('For CVDP information visit: https://www.cesm.ucar.edu/working_groups/CVC/cvdp/')
         print('   ')
 
+    #########
+
+    def derive_variables(self,vars_to_derive=None,ts_dir=None):
+
+        """
+        Derive variables acccording to steps given here.  Since derivations will depend on the 
+        variable, each variable to derive will need its own set of steps below.
```

Comment:
I think this is totally fine for now, but long-term I wonder if we can figure out a way to add the derivation equations to the variable defaults YAML file itself, and then simply have this function properly parse those equations to produce the derived variable.  That way future users won't have to modify the core ADF python code if they want to add a new derived variable.

Thoughts?  @brianpm @justin-richling

Reply:
I like that idea...I am not sure how to implement it though.